File: src/main/java/ec/edu/ups/icc/fundamentos01/categories/mappers/CategoryMapper.java
package ec.edu.ups.icc.fundamentos01.categories.mappers;

import ec.edu.ups.icc.fundamentos01.categories.dto.CategoryResponseDto;
import ec.edu.ups.icc.fundamentos01.categories.entities.CategoryEntity;

public class CategoryMapper {

    public static CategoryResponseDto toResponse(CategoryEntity entity) {
        CategoryResponseDto dto = new CategoryResponseDto();
        dto.setId(entity.getId());
        dto.setName(entity.getName());
        return dto;
    }
}

File: nestjs-practica05/src/categories/mappers/category.mapper.ts
import { CategoryEntity } from '../entities/category.entity';
import { CategoryResponseDto } from '../dtos/category-response.dto';
import { CreateCategoryDto } from '../dtos/create-category.dto';

export class CategoryMapper {
  static toEntityFromDto(dto: CreateCategoryDto): CategoryEntity {
    const entity = new CategoryEntity();
    entity.name = dto.name;
    entity.description = dto.description;
    return entity;
  }

  static toResponse(entity: CategoryEntity): CategoryResponseDto {
    const response = new CategoryResponseDto();
    response.id = entity.id;
    response.name = entity.name;
    response.description = entity.description;
    return response;
  }
}
